src/app/shell/admin-tools/users/users.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { MatTabChangeEvent } from '@angular/material/tabs';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Select, Store } from '@ngxs/store';
+import { Observable, Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, filter, takeUntil } from 'rxjs/operators';
+import { UserTabsUkr, UserTabsUkrReverse } from 'src/app/shared/enum/enumUA/tech-admin/users-tabs';
+import { NoResultsTitle } from 'src/app/shared/enum/no-results';
+import { ChildCards } from 'src/app/shared/models/child.model';
+import { UsersTable } from 'src/app/shared/models/usersTable';
+import { GetChildren } from 'src/app/shared/store/admin.actions';
+import { AdminState } from 'src/app/shared/store/admin.state';
+import { UserState } from 'src/app/shared/store/user.state';
+import { Util } from 'src/app/shared/utils/utils';
+
+@Component({
+  selector: 'app-users',
+  templateUrl: './users.component.html',
+  styleUrls: ['./users.component.scss']
+})
+export class UsersComponent implements OnInit {
+
+  @Select(UserState.isLoading)
+  isLoadingCabinet$: Observable<boolean>;
+  @Select(AdminState.children)
+  children$: Observable<ChildCards>;
+
+  readonly userRoleUkr = UserTabsUkr;
+  readonly noUsers = NoResultsTitle.noUsers;
+
+  filter = new FormControl('');
+  filterValue: string;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  tabIndex: number;
+  allUsers = [];
+  parents = [];
+  children: UsersTable[];
+
+  constructor(
+    public store: Store,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) { }
+
+  ngOnInit(): void {
+    this.filter.valueChanges
+    .pipe(takeUntil(this.destroy$), debounceTime(200), distinctUntilChanged())
```

Comment:
I think it's not possible to have !val not an empty string (I mean is it possible to receive here null/undefined/0/NaN ??? ), right? So you can simplify it to
this.filterValue = val;